src/coreneuron/sim/fadvance_core.cpp

```diff
@@ -366,6 +354,13 @@ static void* nrn_fixed_step_thread(NrnThread* nth) {
             Instrumentor::phase p("update");
             update(nth);
         }
+#ifdef ENABLE_SONATA_REPORTS
+        {
+            Instrumentor::phase p("flush-reports");
+            // Check if is time to flush reports every sim step
+            nrn_flush_reports(nrn_threads[0]._t);
+        }
+#endif
     }
```

Comment:
@jorblancoa: Just to understand and clarify: this branch will be executed by every `NrnThread`. In the case of multiple threads (e.g. multi-cycle run), the first thread might come here and then will call flush but then 2nd thread is not yet completed.

Additionally, this function is executed under OpenMP threads (e.g. when we enable coreneuron openmp) and have multi-cycle run execution.

Have you tested above scenarios or know if they are supposed to work?

Reply:
I dont know how to test those things, but tested in several simulations without issue.
Also it will check if it is time to write, buf it a thread didnt record its node_ids, it wont flush since the buffer is not full.

If you tell me where to move the call so it is called every simulation step without the multithreading or how to test those things i will happily do so.